src/cmdstan/command_helper.hpp

```diff
@@ -634,6 +670,190 @@ void services_log_prob_grad(const stan::model::model_base &model, bool jacobian,
     output_stream << gradients.back() << "\n";
   }
 }
+/**
+ * Return true if sampler config is adaptive NUTS-HMC with non-unit metric
+ * otherwise false.
+ *
+ * @param sample_arg user config
+ * @return int num chains or paths
+ */
+bool allow_multichain(argument_parser &parser) {
+  auto user_method = parser.arg("method");
+  if (user_method->arg("pathfinder"))
+    return true;
+  if (!user_method->arg("sample"))
+    return false;
+  auto sample_arg = parser.arg("method")->arg("sample");
+  categorical_argument *adapt
+      = dynamic_cast<categorical_argument *>(sample_arg->arg("adapt"));
+  bool adapt_engaged
+      = dynamic_cast<bool_argument *>(adapt->arg("engaged"))->value();
+  list_argument *algo
+      = dynamic_cast<list_argument *>(sample_arg->arg("algorithm"));
+  if (algo->value() == "hmc") {
+    list_argument *engine
+        = dynamic_cast<list_argument *>(algo->arg("hmc")->arg("engine"));
+    list_argument *metric
```

Comment:
The second half of this function is repeating the logic of the `validate_multi_chain_config` function but just returns a boolean rather than raising an error. We should only be doing this one place